modules/login/php/requestaccount.class.inc

```diff
@@ -269,7 +269,67 @@ class RequestAccount extends \NDB_Form
             }
         }
 
+        // Verify reCAPTCHA
+        $err = [];
+        if (isset($values['g-recaptcha-response']) && isset($reCAPTCHAPrivate)) {
+            $recaptcha = new \ReCaptcha\ReCaptcha($reCAPTCHAPrivate);
+            $resp      = $recaptcha->verify(
+                $values['g-recaptcha-response'],
+                $_SERVER['REMOTE_ADDR']
+            );
+            if (!$resp->isSuccess()) {
+                $err['captcha'] = 'Please complete the reCaptcha!';
+            }
+        }
         // Show success message even if email already exists for security reasons
         $this->tpl_data['success'] = true;
+
+        // If there are error messages, return early without sending emails.
+        if (!empty($err)) {
+            return;
+        }
+
+        // Send an email to all users meeting the following criteria:
+        //     - Associated with the Site
+        //     - Associated with the Project
+        //     - Have the User Management permission
+        $query  = <<<SQL
+        SELECT DISTINCT u.Email,u.userID 
+        FROM users u 
+        JOIN user_psc_rel upr 
+        ON u.ID = upr.UserID 
+        JOIN user_project_rel uprojr 
+        ON u.ID = uprojr.UserID 
+        WHERE upr.CenterID=:center
+        AND uprojr.ProjectID=:project
+SQL;
```

Comment:
```suggestion
        $query  = "SELECT DISTINCT u.Email,u.userID
             FROM users u
                JOIN user_psc_rel upr
                    (ON u.ID = upr.UserID)
               JOIN user_project_rel uprojr
                   (ON u.ID = uprojr.UserID)
            WHERE upr.CenterID=:center
                AND uprojr.ProjectID=:project"
```

Formatting SQL isn't a good use of HEREDOCs and also the formatting is hard to read.